Simplenote/src/androidTest/java/com/automattic/simplenote/integration/TagDialogFragmentTest.kt

```diff
@@ -0,0 +1,326 @@
+import org.junit.Assert.*
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4ClassRunner::class)
+@MediumTest
+class TagDialogFragmentTest : BaseUITest() {
+
+    @Test
+    fun editTagWithValidName() {
+        createTag("tag1")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag1"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag1")
+
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText("tag5"))
+        onView(allOf(withText("tag5"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(isDisplayed())).perform(click())
+
+        val tag1 = getTag("tag1")
+        val tag5 = getTag("tag5")
+
+        assertNull(tag1)
+        assertNotNull(tag5)
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(note.tags, listOf("tag5"))
+    }
+
+    @Test
+    fun editTagTooLong() {
+        createTag("tag1")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag1"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag1")
+
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        val longName = getRandomString(258)
+        onView(allOf(withText("tag1"), isDescendantOfA(withId(R.id.input_tag_name))))
+                .perform(ViewActions.replaceText(longName))
+
+        val tooLongMessage = getResourceString(R.string.tag_error_length)
+        onView(withId(R.id.input_tag_name)).check(matches(hasTextInputLayoutErrorText(tooLongMessage)))
+
+        val saveText = getResourceString(R.string.save)
+        onView(withText(saveText)).inRoot(isDialog()).check(matches(not(isEnabled())))
+        val cancelText = getResourceString(R.string.cancel)
+        onView(withText(cancelText)).inRoot(isDialog()).perform(click())
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(note.tags, listOf("tag1"))
+    }
+
+    @Test
+    fun editTagWithSpace() {
+        createTag("tag10")
+        // To edit tags, tags should belong a note
+        val note = createNote("Hello World", listOf("tag10"))
+
+        assertEquals(tagsBucket.count(), 1)
+        assertEquals(notesBucket.count(), 1)
+
+        launchFragment("tag10")
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name)))).check(matches(isDisplayed()))
+
+        onView(allOf(withText("tag10"), isDescendantOfA(withId(R.id.input_tag_name))))
```

Comment:
Suggestion (💡): Just like you did with the `val longName = getRandomString(258)` test above, what if we made this explicit as well, by extracting a variable: `val tagWithSpace = "tag 3"`

PS: Same applies for the potential `val emptyTag = ""` variable on the test below and any other such variable that can help with making the testing part explicit.

Reply:
Fixed in 0627e2a5a35b912fa67d8ecb20127df887f6c6ee. 

PS: I change a field visibility in BaseUITest and accidentally went into this commit.

Reply:
> PS: I change a field visibility in BaseUITest and accidentally went into this commit.

All good... 😅